src/control/Snapshot/Snapshot.js

```diff
@@ -0,0 +1,140 @@
+    print.title = 'Print snapshot';
+    print.className = 'print';
+    print.addEventListener('click', this.printSnapshot.bind(this));
+    element.appendChild(print);
+  }
+
+  /**
+   * Callback to deactivate the snapshot control.
+   * @private
+   */
+  deactivate() {
+    this.element.classList.remove('active');
+  }
+
+  /**
+   * Callback for the snapshot button click event.
+   * @param {MouseEvent} event The event to handle
+   * @private
+   */
+  captureImage(event) {
+    event.preventDefault();
+
+    // Create a new canvas element to combine multiple map canvas data to.
+    const outputCanvas = document.createElement('canvas');
+    const [width, height] = this.getMap().getSize();
+    outputCanvas.width = width;
+    outputCanvas.height = height;
+    const outputContext = outputCanvas.getContext('2d');
+
+    // Draw each canvas from this map into the new canvas.
+    // Logic for transforming and drawing canvases derived from ol export-pdf example.
+    // https://github.com/openlayers/openlayers/blob/6f2ca3b9635f273f6fbddab834bd9126c7d48964/examples/export-pdf.js#L61-L85
+    Array.from(this.getMap().getTargetElement().querySelectorAll('.ol-layer canvas'))
+      .filter(canvas => canvas.width > 0)
+      .forEach((canvas) => {
+        const { opacity } = canvas.parentNode.style;
+        outputContext.globalAlpha = opacity === '' ? 1 : Number(opacity);
+
+        // Get the transform parameters from the style's transform matrix.
+        // This is necessary so that vectors align with raster layers.
+        const { transform } = canvas.style;
+        const matrix = transform
+          .match(/^matrix\(([^(]*)\)$/)[1]
+          .split(',')
+          .map(Number);
+
+        // Apply the transform to the export map context.
+        CanvasRenderingContext2D.prototype.setTransform.apply(
+          outputContext,
+          matrix,
+        );
+        outputContext.drawImage(canvas, 0, 0);
+      });
```

Comment:
What do you think about this @symbioquine ? Implementing the transform + draw logic in a more declarative way. Also linked to the reference - I was torn because as a whole the example is overly complex and we only need this portion - but still good to reference :+1: